docs/source/tools/trigger-service/auth0_example.rst

```diff
@@ -0,0 +1,286 @@
+.. Copyright (c) 2021 Digital Asset (Switzerland) GmbH and/or its affiliates. All rights reserved.
+.. SPDX-License-Identifier: Apache-2.0
+
+Auth0 Example Configuration
+###########################
+
+This section describes a minimal example configuration of the trigger service with authorization enabled
+using `Auth0 <auth0_>`_ as the OAuth 2.0 provider together with the OAuth 2.0 middleware included in Daml Connect.
+It uses the sandbox as the Daml ledger.
+
+Configure Auth0
+~~~~~~~~~~~~~~~
+
+Sign up for an account on Auth0 to follow this guid.
+
+Create an API
+*************
+
+First, `create a new API <auth0-create-api_>`_ on the Auth0 API dashboard.
+This will represent the Daml ledger API and controls properties of access tokens issued for the ledger API.
+
+- Enter the name of the API, e.g. ``ex-daml-api``.
+- Enter the API identifier: ``https://daml.com/ledger-api``.
+- Select the signing algorithm ``RS256``.
+- Press the "create" button.
+
+Enter the `settings <auth0-api-settings_>`_ of the newly created API.
+
+- Allow offline access in the access settings section to enable issuance of refresh tokens.
+
+Create an Application
+*********************
+
+`Create a new native application <auth0-create-application_>`_.
+This will represent the OAuth 2.0 middleware.
+
+- Enter the name of the application, e.g. ``ex-daml-auth-middleware``.
+- Choose the application type "native".
+- Press the "create" button.
+
+Enter the `settings <auth0-application-settings_>`_ of the newly created application.
+
+- Configure the allowed callback URLs: ``http://localhost:5000/auth/cb``.
+
+  This is the URL to the callback endpoint of the auth middleware, in this case through the reverse proxy.
+- Take note of the "Client ID" and "Client Secret" displayed in the "Basic Information" section.
+- Take note of the following URLs in the "Endpoints" tab of the advanced settings:
+    - OAuth Authorization URL,
+    - OAuth Token URL, and
+    - JSON Web Key Set.
+
+Create a Rule
+*************
+
+`Create a new rule <auth0-create-rule_>`_.
```

Comment:
We should add a disclaimer here that this does not validate that the user is actually authorized to act as those parties and that in a real setup, there needs to be a mapping from users to the list of allowed claims somewhere.

Reply:
:+1: I've moved https://github.com/digital-asset/daml/pull/8622/files/cd09bad242cd796fb0be12939ead38385c190760#diff-e9c85b993f267409dd5b9c6a5db5d817e4037e61f15e1efbc9fc1040a0b82f39R111-R112 up and expanded it a bit.